xdsl/irdl/declarative_assembly_format.py

```diff
@@ -438,27 +438,41 @@ def parse(self, parser: Parser, state: ParsingState) -> None:
 
     def print(self, printer: Printer, state: PrintingState, op: IRDLOperation) -> None:
         if self.print_properties:
-            if (
-                not (set(op.attributes.keys()) | set(op.properties.keys()))
-                - self.reserved_attr_names
-            ):
-                return
             if any(name in op.attributes for name in op.properties):
                 raise ValueError(
                     "Cannot print attributes and properties with the same name "
                     "in a signle dictionary"
```

Comment:
Can you fix this typo while you're at it? :upside_down_face: 
```suggestion
                    "in a single dictionary"
```